src/node/rpc/managementfrontend.h

```diff
@@ -26,7 +26,13 @@ namespace ccf
         const auto in = args.params.get<JoinNetwork::In>();
         node.join_network(args.rpc_ctx, in);
 
-        return jsonrpc::success();
+        // TODO: Would be preferable to make this a jsonrpc::no_response, give
+        // the return type an enum rather than bool, make it clear that this
+        // handler is not returning a value - something else is in the future.
+        return jsonrpc::error(
```

Comment:
Why is this returning an error?

Reply:
This return value is never actually returned to the user (because `join_network` has set `is_pending` on the `rpc_ctx`). This means its different from every other method where we can look at how `params` is parsed and what's returned to get the input/output schema. I want to make it clear that this is an exceptional case, so it has a self-descriptive message. This could be a `jsonrpc::success` containing a string, but then if something breaks and this _does_ reach the user we're in real trouble (we've returned something of the wrong type, that looks like a result but is a failure). So if _this return value_ is ever seen, its an error.

As the comment says, this is a temporary - I think we can/should roll `is_pending` into the return value and make this look less weird.

Reply:
thanks for the explanation